lib/elixir/test/elixir/kernel/typespec_test.exs

```diff
@@ -884,4 +891,18 @@ defmodule Kernel.TypespecTest do
       end
     end
   end
+
+  test "warn on discouraged types" do
+    message = capture_err(fn ->
```

Comment:
I think it would be better to call `capture_io(:stderr, fn ->` directly, we don't save a lot with the `capture_err` function.